src/docs/arch-rts.dox

```diff
@@ -110,4 +109,154 @@ Update list: U3 (30)\n
+
+History store deletions are done using a standard history store cursor, starting at the highest
+possible start timestamp, and removing keys from newest to oldest until we hit an update with a
+start timestamp larger than the rollback timestamp.
+
+Note: in-memory databases don't have a history store, so none of this section will apply.
+
+@section iterating-over-b-trees Iterating over B-Trees
+
+This is one of the simplest parts of the RTS subsystem, as a metadata cursor is specifically
+designed for these sorts of "iterate over a B-Tree" operations. We use this functionality for RTS,
+with the caveat that we tweak a few options so that corrupt or missing files don't raise errors. The
+reasoning here is that RTS is orthogonal to salvage, and might be called as part of salvage, so we
+simply need to "do our best". We do log the issues we find, however.
+
+Less obviously, the history store needs one final inspection regardless of whether any other B-Trees
+were touched. This is because we usually rely on getting to the history store "via" the data store,
+but if we don't touch a B-Tree, that doesn't mean there aren't updates in the history store after
+the stable timestamp.
+
+@section iterating-over-pages Iterating over pages
+
+RTS tries to avoid looking at a page if it can. There are two ways we can avoid instantiating a
+page: if it was truncated at/before the stable timestamp, and if the page state is anything other
+than on-disk. If neither of these are satisfied, we need to instantiate the page to look at it, but
+that doesn't mean we look at the contents. Rather, we use the existing tree walk custom skipping
+functionality to check the page metadata, stuff like reconciliation results. These checks are fairly
+low-level, so they're outside the scope of this document.
+
+One final note is that we deliberately do not call RTS for internal pages, as they do not have
+updates. They are still read into the cache in order to navigate to their leaf pages.
+
+@section iterating-over-updates Iterating over updates
+
+While the details of what happens here depends on the page type (row-store, VLCS or FLCS), the
+overall idea is the same: we start by looking at the insert list (i.e. any items inserted before the
+first key on the page), iterate over all of the keys on the page, then look at the append list
+(i.e. any items inserted after the last key on the page). We mark the page as dirty so that we
+reconcile it in future.
+
+The mechanics of deleting items (or inserting tombstones) from these places are quite involved, and
+out of scope for the architecture guide.
+
+@section interaction-with-checkpoints-and-eviction Interaction with checkpoints and eviction
+
+Checkpoints cannot be performed while RTS is running, even if they start before RTS. This is part of
+the contract that the system must be in a quiescent state before running RTS, but it's also enforced
+by taking the session's checkpoint lock.
```

Comment:
checkpoint lock is present in the connection. Change it to `connection`.

Reply:
Thanks - fixed in 5d27bf2.